common/bytes.go

```diff
@@ -58,12 +56,38 @@ func CopyBytes(b []byte) (copiedBytes []byte) {
 
 func HasHexPrefix(str string) bool {
 	l := len(str)
-	return l >= 2 && str[0:2] == "0x"
+	return l >= 2 && (str[0:2] == "0x" || str[0:2] == "0X")
+}
+
+func isHexCharacter(c byte) bool {
+	switch {
+	case '0' <= c && c <= '9':
+		return true
+	case 'a' <= c && c <= 'f':
+		return true
+	case 'A' <= c && c <= 'F':
+		return true
+	}
+	return false
```

Comment:
Please use a boolean expression instead of switch.

Reply:
The switch is taken from the `hex` package. I think it's more readable. If you insist, I could change it...